lib/omnibus/git_cache.rb

```diff
@@ -22,6 +22,18 @@ class GitCache
     include Util
     include Logging
 
+    # The serial number represents compatibility of a cache entry with the
+    # current version of the omnibus code base. Any time a change is made to
+    # omnibus that makes the code incompatible with any cache entries created
+    # before the code change, the serial number should be incremented.
+    #
+    # For example, if a code change generates content in the `install_dir`
+    # before cache snapshots are taken, any snapshots created before upgrade
+    # will not have the generated content, so these snapshots would be
+    # incompatible with the current omnibus codebase. Incrementing the serial
+    # number ensures these old shapshots will not be used in subsequent builds.
+    SERIAL_NUMBER = 1
```

Comment:
As the git caching is a hard to understand beast, I'm worried contributors won't know the appropriate time to bump this number. Would it make more sense to base the cache busting off the `MAJOR.MINOR` of `Omnibus::VERSION `?